string.c

```diff
@@ -2707,11 +2709,11 @@ str_uplus(VALUE str)
 
 /*
  * call-seq:
- *   -str  -> str (frozen)
+ *   -string -> frozen_string
  *
- * Returns a frozen, possibly pre-existing copy of the string.
+ * Returns a frozen duplicate of +self+, which may have already existed.
```

Comment:
I don't think this is accurate.  It can return self if self is frozen and their is no pre-existing copy of the string.

Reply:
Fixed.